diff --git a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
index 10e1ea16..f4067f69 100644
--- a/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
+++ b/internal/frontend/bridge-gui/bridge-gui/DeployLinux.cmake
@@ -16,6 +16,10 @@
 # along with Proton Mail Bridge. If not, see <https://www.gnu.org/licenses/>.
 
 cmake_minimum_required(VERSION 3.22)
+#*****************************************************************************************************************************************************
+# International Components for Unicode (ICU)
+#*****************************************************************************************************************************************************
+find_package(ICU 56.0 COMPONENTS i18n uc data REQUIRED)
 
 #*****************************************************************************************************************************************************
 # Deploy
@@ -54,9 +58,9 @@ AppendQt6Lib("libQt6Gui.so.6")
 AppendQt6Lib("libQt6Core.so.6")
 AppendQt6Lib("libQt6QuickTemplates2.so.6")
 AppendQt6Lib("libQt6DBus.so.6")
-AppendQt6Lib("libicui18n.so.56")
-AppendQt6Lib("libicuuc.so.56")
-AppendQt6Lib("libicudata.so.56")
+AppendQt6Lib("libicui18n.so")
+AppendQt6Lib("libicuuc.so")
+AppendQt6Lib("libicudata.so")
 AppendQt6Lib("libQt6XcbQpa.so.6")
 AppendQt6Lib("libQt6WaylandClient.so.6")
 AppendQt6Lib("libQt6WlShellIntegration.so.6")
