diff --git a/internal/frontend/bridge-gui/bridge-gui-tester/GRPCService.cpp b/internal/frontend/bridge-gui/bridge-gui-tester/GRPCService.cpp
index 22d02708..65995937 100644
# make the test pass by changing the order of the title and url
--- a/internal/frontend/bridge-gui/bridge-gui-tester/GRPCService.cpp
+++ b/internal/frontend/bridge-gui/bridge-gui-tester/GRPCService.cpp
@@ -365,8 +365,8 @@ grpc::Status GRPCService::RequestKnowledgeBaseSuggestions(ServerContext*, String
     QList<bridgepp::KnowledgeBaseSuggestion> suggestions;
     for (qsizetype i = 1; i <= 3; ++i) {
         suggestions.push_back(        {
-            .title = QString("Suggested link %1").arg(i),
             .url = QString("https://proton.me/support/bridge#%1").arg(i),
+            .title = QString("Suggested link %1").arg(i),
         });
     }
     qtProxy_.sendDelayedEvent(newKnowledgeBaseSuggestionsEvent(app().mainWindow().knowledgeBaseTab().getSuggestions()));
