diff --git a/Makefile b/Makefile
index 22f6e279..4164683e 100644
# fix typo that causes "warning: no packages being tested depend on matches for pattern" issue
# add parallelism to tests to speed up CI
# add -json flag to tests to make output parseable
--- a/Makefile
+++ b/Makefile
@@ -233,24 +233,24 @@ add-license:
 change-copyright-year:
 	./utils/missing_license.sh change-year
 
-GOCOVERAGE=-covermode=count -coverpkg=github.com/ProtonMail/proton-bridge/v3/internal/...,github.com/ProtonMail/proton-bridge/v3/pkg/...,
+GOCOVERAGE=-covermode=count -coverpkg=github.com/ProtonMail/proton-bridge/v3/internal/...,github.com/ProtonMail/proton-bridge/v3/pkg/...
 GOCOVERDIR=-args -test.gocoverdir=$$PWD/coverage
 
 test: gofiles
-	mkdir -p coverage/unit-${GOOS}
-	go test \
-		-v -timeout=20m -p=1 -count=1 \
+	@mkdir -p coverage/unit-${GOOS}
+	@go test \
+		-json -timeout=20m -p=4 -count=1 \
 		${GOCOVERAGE} \
 		-run=${TESTRUN} ./internal/... ./pkg/... \
 		${GOCOVERDIR}/unit-${GOOS}
 
 test-race: gofiles
-	go test -v -timeout=40m -p=1 -count=1 -race -failfast -run=${TESTRUN} ./internal/... ./pkg/...
+	@go test -json -timeout=40m -p=4 -count=1 -race -failfast -run=${TESTRUN} ./internal/... ./pkg/...
 
 test-integration: gofiles
-	mkdir -p coverage/integration
-	go test \
-		-v -timeout=60m -p=1 -count=1 -tags=test_integration \
+	@mkdir -p coverage/integration
+	@go test \
+		-json -timeout=60m -p=4 -count=1 -tags=test_integration \
 		${GOCOVERAGE} \
 		github.com/ProtonMail/proton-bridge/v3/tests \
 		${GOCOVERDIR}/integration
@@ -260,23 +260,23 @@ test-integration-debug: gofiles
 	dlv test github.com/ProtonMail/proton-bridge/v3/tests -- -test.v -test.timeout=10m -test.parallel=1 -test.count=1
 
 test-integration-race: gofiles
-	go test -v -timeout=60m -p=1 -count=1 -race -failfast github.com/ProtonMail/proton-bridge/v3/tests
+	@go test -json -timeout=60m -p=4 -count=1 -race -failfast github.com/ProtonMail/proton-bridge/v3/tests
 
 test-integration-nightly: gofiles
-	mkdir -p coverage/integration
-	go test \
-		-v -timeout=90m -p=1 -count=1 -tags=test_integration \
+	@mkdir -p coverage/integration
+	@go test \
+		-json -timeout=90m -p=4 -count=1 -tags=test_integration \
 		${GOCOVERAGE} \
 		github.com/ProtonMail/proton-bridge/v3/tests \
 		${GOCOVERDIR}/integration \
 		nightly
 
 fuzz: gofiles
-	go test -fuzz=FuzzUnmarshal 	 -parallel=4 -fuzztime=60s $(PWD)/internal/legacy/credentials
-	go test -fuzz=FuzzNewParser 	 -parallel=4 -fuzztime=60s $(PWD)/pkg/message/parser
-	go test -fuzz=FuzzReadHeaderBody -parallel=4 -fuzztime=60s $(PWD)/pkg/message
-	go test -fuzz=FuzzDecodeHeader 	 -parallel=4 -fuzztime=60s $(PWD)/pkg/mime
-	go test -fuzz=FuzzDecodeCharset  -parallel=4 -fuzztime=60s $(PWD)/pkg/mime
+	@go test -json -fuzz=FuzzUnmarshal 	 -parallel=4 -fuzztime=20s $(PWD)/internal/legacy/credentials
+	@go test -json -fuzz=FuzzNewParser 	 -parallel=4 -fuzztime=20s $(PWD)/pkg/message/parser
+	@go test -json -fuzz=FuzzReadHeaderBody -parallel=4 -fuzztime=20s $(PWD)/pkg/message
+	@go test -json -fuzz=FuzzDecodeHeader 	 -parallel=4 -fuzztime=20s $(PWD)/pkg/mime
+	@go test -json -fuzz=FuzzDecodeCharset  -parallel=4 -fuzztime=20s $(PWD)/pkg/mime
 
 bench:
 	go test -run '^$$' -bench=. -memprofile bench_mem.pprof -cpuprofile bench_cpu.pprof ./internal/store
